### ISSUE TITLE
Refuses to be added to Login Items

### ISSUE THREAD

**Issue by rangedsp** [pilotmoon/Scroll-Reverser#165]
Model Name:	        MacBook Pro
Model Identifier:	Mac14,6
Chip:	                 Apple M2 Max
System Version:	macOS 13.3.1 (22E261)

Repro steps: 
1. Download fresh copy of scroll-reverser 1.8.2
2. Unzip
3. Drag into Applications
4. Launch app
5. Goto Login Items
6. Under "Open at Login", hit "+"
7. Select "Scroll Reverser", hit "Open"
8. Observe that it doesn't get added to the list

Very weird, I've never had an issue with this, worked on an Intel mac (running Monteray)

**Comment by paultheman** [comment]
I have the same issue on macOS 13.3.1.

**Comment by pilotmoon** [comment]
If this is happening, try adding it manually to the Login items list instead. Also ensure that it is in the Applications folder, and not in any other location. And that there are not any duplicate copies of the app.

**Comment by pilotmoon** [comment]
This would appear to be a macOS issue.

**Comment by ZyntaaX** [comment]
This is a problem for me as well on the M2 Pro. The app is in the /Applications folder, no other location. When i press the "+" button and select the application, nothing happens. No errors, nothing, it just won't add to the login items list.

Any fix in sight?

**Comment by pilotmoon** [comment]
The login items list is part of the macOS settings. I don't know how you expect me to fix it.

**Comment by rangedsp** [comment]
@pilotmoon I am unsure about other, but my repro steps is specifically about adding Scroll Reverser to the login items manually.

Looking at the [function you are using to register on launch](https://github.com/pilotmoon/Scroll-Reverser/blob/master/LauncherController.m), I think I've figured out the reason: `SMLoginItemSetEnabled` has been deprecated as of Ventura, [as per Apple docs](https://developer.apple.com/documentation/servicemanagement/1501557-smloginitemsetenabled). 

My current workaround is to create an automator job that launches Scroll Reverser, and add that job to login items instead.

**Comment by pilotmoon** [comment]
Ah, I see. SMLoginItemSetEnabled still works for me, on Ventura, even though deprecated. It adds the app to the "Allow in the Background" section below the login items list. Perhaps the settings dialog is programmed not to add apps to the manual login items list that are already in the allow in the background section.  Was the normal mechanism not working for you?

**Comment by rangedsp** [comment]
@pilotmoon no, the normal mechanism didn't work, that's why I tried to add the app manually.

> Perhaps the settings dialog is programmed not to add apps to the manual login items list that are already in the allow in the background section

So this is really weird, Zoom, a couple of Logitech apps, and MS Teams shows up in the 'allow in background' section, and I was able to manually add it to the 'open at login' section. BUT, DisplayLink Manager, Hidden Bar, Facebook Messenger etc can't be added just like Scroll Reverser. 

I am wondering if it's to do with calling the deprecated API that causes this issue? 
(if I have some time to learn mac development I might take a crack at this)

**Comment by theotherlinh** [comment]
Dunno if it helps any debugging, but I'm having this issue as well. In trying to reinstall and try from scratch, I also noticed Scroll Reverser could not be removed from the "Allow in background" list.  I checked /Library/LaunchAgents and LauchDaemons as well as ~/Library/Launch* and nothing was listed.

Also just using Automator as a workaround.

**Comment by com314159** [comment]
me too, I use Automator to start scroll reverser now

**Comment by londbell** [comment]
same issue,ventura(m1 pro)

**Comment by TyJaYo** [comment]
I was having this issue with a different application (Magnet) and was able to resolve it by dragging the app to the Dock, then control-clicking it and selecting "Options" > "Open at Login". The app now appears on the "Login items" "Open at Login" list and stays there. Perhaps it would work for this app as well.

**Comment by swtprsn** [comment]
> I was having this issue with a different application (Magnet) and was able to resolve it by dragging the app to the Dock, then control-clicking it and selecting "Options" > "Open at Login". The app now appears on the "Login items" "Open at Login" list and stays there. Perhaps it would work for this app as well.

It works. Thank you.

**Comment by rosschapman** [comment]
I was having a similar issue with Ventura on an Intel Mac where I wasn't able to add or remove login items. I would click the ➕ or ➖ icons and nothing would happen. The dialog window would not even open. However I just fixed this by going into my Privacy and Security settings panel and clicking the `Turn Off` FileVault button. I didn't proceed with deactivation, but for some reason when I closed the dialog window, and returned to the Login Items settings page, I was able to add and remove login items per normal. Perhaps it's just a strange coincidence?

**Comment by pilotmoon** [comment]
I have now implemented the new SMAppService API for start at login. This will be released in the next update.

**Comment by pilotmoon** [comment]
To all who reported this "Start at login" issue, it should now be fixed in [Build 10519](https://github.com/pilotmoon/Scroll-Reverser/releases/tag/b10519), which now requires macOS 13 (Ventura) minimum.

It is available by in-app update, with the beta option enabled.

If you get a chance to try it, please comment to let me know if it has solved this issue for you.

**Comment by pilotmoon** [comment]
Now shipped in Scroll Reverser 1.9
